ora/Common/Representables/WindowAccessor.swift

```diff
@@ -13,51 +13,65 @@ struct WindowAccessor: NSViewRepresentable {
         Coordinator(self)
     }
 
+    @MainActor
     class Coordinator {
         var parent: WindowAccessor
-        var observers: [Any] = []
+        var observationTasks: [Task<Void, Never>] = []
 
         init(_ parent: WindowAccessor) {
             self.parent = parent
         }
 
-        @objc func didEnterFullScreen(_ notification: Notification) {
+        func didEnterFullScreen(_ notification: Notification) {
             guard let window = notification.object as? NSWindow else { return }
             parent.isFullscreen = true
             parent.updateTrafficLights(for: window)
         }
 
-        @objc func didExitFullScreen(_ notification: Notification) {
+        func didExitFullScreen(_ notification: Notification) {
             guard let window = notification.object as? NSWindow else { return }
             parent.isFullscreen = false
             parent.updateTrafficLights(for: window)
         }
     }
 
-    func makeNSView(context: Context) -> NSView {
-        let view = NSView()
+    final class WindowView: NSView {
+        var onMoveToWindow: ((NSWindow) -> Void)?
+
+        override func viewDidMoveToWindow() {
+            super.viewDidMoveToWindow()
+            if let window {
+                onMoveToWindow?(window)
+            }
+        }
+    }
 
-        DispatchQueue.main.async {
-            guard let window = view.window else { return }
+    func makeNSView(context: Context) -> NSView {
+        let view = WindowView()
+        view.onMoveToWindow = { window in
             isFullscreen = window.styleMask.contains(.fullScreen)
 
             let coordinator = context.coordinator
 
-            let enterObserver = NotificationCenter.default.addObserver(
-                forName: NSWindow.didEnterFullScreenNotification,
-                object: window,
-                queue: nil,
-                using: coordinator.didEnterFullScreen
-            )
-
-            let exitObserver = NotificationCenter.default.addObserver(
-                forName: NSWindow.didExitFullScreenNotification,
-                object: window,
-                queue: nil,
-                using: coordinator.didExitFullScreen
-            )
-
-            coordinator.observers = [enterObserver, exitObserver]
+            let didEnterFullScreenNotificationTask = Task {
```

Comment:
If the coordinator is an NSView, could it subscribe to these things with the selector version of the calls? In my experience, these long-running tasks observing sequences make it very easy to introduce retain cycles (as is happening in this right now)

Coordinator retains the task which retains the closure which retains the coordinator